src/creature.cpp

```diff
@@ -176,7 +176,8 @@ bool Creature::sees( const tripoint &t, bool is_player ) const
     const int range_night = sight_range( 0 );
     const int range_max = std::max( range_day, range_night );
     const int range_min = std::min( range_cur, range_max );
-    const int wanted_range = rl_dist( pos(), t );
+    // If monster sees player is checked wanted range is affected by a modifier
+    const int wanted_range = rl_dist( pos(), t ) / ( is_player ? g->u.get_sees_modifier() : 1.0f );
```

Comment:
You can pass the `sees_modifer` as an argument from above - this will make the whole thing much cleaner and allow generalizing stealth to other critters.